src/ReachSets/ContinuousPost/BFFPS19/reach_blocks.jl

```diff
@@ -0,0 +1,269 @@
+
+
+#helper functions
+@inline function deco_post_sparse(b, blocks, Whatk_blocks, partition,
+                                  ϕpowerk, Xhat0, output_function)
+    for i in 1:b
+        bi = partition[blocks[i]]
+        Xhatk = ZeroSet(length(bi))
+        for (j, bj) in enumerate(partition)
+          block = ϕpowerk[bi, bj]
+          if !iszero(block)
+              Xhatk = Xhatk + block * Xhat0[j]
+          end
+        end
+        Xhatk_lazy = (U == nothing ? Xhatk : Xhatk + Whatk_blocks[i])
+        Xhatk_d[i] = (output_function == nothing) ?
+          overapproximate(blocks[i], Xhatk_lazy) :
+          Xhatk_lazy
+    end
+
+    array_d = CartesianProductArray(copy(Xhatk_d))
+
+    X_store = (output_function == nothing) ?
+                  array_d :
+                  box_approximation(output_function(array_d))
+
+    return X_store
+end
+
+@inline function deco_post_dense(b, blocks, Whatk_blocks, partition,
+                                  ϕpowerk, arr::Vector{LazySet{NUM}}, arr_length, U, Xhat0, Xhatk, output_function) where {NUM}
+    for i in 1:b
+        bi = partition[blocks[i]]
+        for (j, bj) in enumerate(partition)
+            arr[j] = ϕpowerk[bi, bj] * Xhat0[j]
+        end
+        if U != nothing
+            arr[arr_length] = Whatk_blocks[i]
+        end
+        Xhatk[i] = (output_function == nothing) ?
```

Comment:
@schillic Can you please check why this line doesn't work inside of this function? It is basically the same code and inputs for this line (I have checked when I debugged) are the same as it was before, when this code was outside. I am really confused here...

Reply:
What does *doesn't work* mean?

Reply:
```
ERROR: MethodError: no method matching overapproximate(::Int64, ::MinkowskiSumArray{Float64,LazySet{Float64}})
Stacktrace:
 [1] deco_post_dense(::Int64, ::Array{Int64,1}, ::Array{LazySet{Float64},1}, ::Array{UnitRange{Int64},1}, ::Array{Float64,2}, ::Array{LazySet{Float64},1}, ::Int64, ::ConstantInput{MinkowskiSum{Float64,LinearMap{Float64,Singleton{Float64,Array{Float64,1}},Float64,SparseMatrixCSC{Float64,Int64}},Hyperrectangle{Float64}}}, ::Array{LazySet{Float64},1}, ::Array{LazySet{Float64},1}, ::Nothing) at /Users/kpotomkin/.julia/dev/Reachability/src/ReachSets/ContinuousPost/BFFPS19/reach_blocks.jl:39
```

Reply:
Ah, you have to pass the `overapproximate` function. It is an argument to `reach_blocks!`.

Reply:
Ahhh true, thank you!